[DOCS] Reorganize README (#900)
elastic/elasticsearch-ruby: README.md

# Elasticsearch

This repository contains Ruby integrations for [Elasticsearch](https://www.elastic.co/products/elasticsearch).

The [`elasticsearch`](https://github.com/elasticsearch/elasticsearch-ruby/tree/master/elasticsearch)
library is a wrapper for two separate libraries:

* [`elasticsearch-transport`](https://github.com/elasticsearch/elasticsearch-ruby/tree/master/elasticsearch-transport),

>   which provides a low-level Ruby client for connecting to an [Elasticsearch](https://www.elastic.co/products/elasticsearch) cluster

Reviewer: I don't think this link is needed. We linked to the same page with the same anchor text earlier.
```suggestion
  which provides a low-level Ruby client for connecting to an Elasticsearch cluster
```

* [`elasticsearch-api`](https://github.com/elasticsearch/elasticsearch-ruby/tree/master/elasticsearch-api),
  which provides a Ruby API for the Elasticsearch RESTful API

```ruby
require 'elasticsearch'

client = Elasticsearch::Client.new log: true

# if you specify Elasticsearch host
# client = Elasticsearch::Client.new url: 'http://localhost:9200', log: true

client.transport.reload_connections!

client.cluster.health

client.search q: 'test'

# etc.
```

Both of these libraries are extensively documented.
**Please read the [`elasticsearch-transport`](http://rubydoc.info/gems/elasticsearch-transport) and the [`elasticsearch-api`](http://rubydoc.info/gems/elasticsearch-api) documentation carefully.**

See also [`doc/examples`](https://github.com/elastic/elasticsearch-ruby/blob/master/docs/examples/README.md) for some practical examples.

> _Keep in mind, that for optimal performance, you should use a HTTP library which supports persistent
> ("keep-alive") connections, e.g. [Patron](https://github.com/toland/patron) or [Typhoeus](https://github.com/typhoeus/typhoeus)._ These libraries are not dependencies of the elasticsearch gems, so be sure to define a dependency in your own application.

Reviewer: Italics are bad for readability generally. I'd use bold to draw attention to the text. I also think you can remove `Keep in mind...`.

I think Elasticsearch would be capitalized here. Otherwise, it should be formatted as a literal. 
I'd also split the second sentence to make it a little more readable.
```suggestion
**For optimal performance, you should use a HTTP library which supports persistent
("keep-alive") connections, e.g. [Patron](https://github.com/toland/patron) or [Typhoeus](https://github.com/typhoeus/typhoeus).** These libraries are not dependencies of the Elasticsearch gems. Ensure you define a dependency for a HTTP library in your own application.
```

This repository contains these additional Ruby libraries:

* [`elasticsearch-extensions`](https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-extensions),
   which provides a set of extensions to the base library,
* [`elasticsearch-dsl`](https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-dsl),
  which provides a Ruby API for the [Elasticsearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html),
* [`elasticsearch-xpack`](https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-xpack),

>   which provides Ruby API for the  [_X-Pack_](https://www.elastic.co/products/x-pack) APIs. This API is going to be merged into `elasticsearch-api` on v8.0.

Reviewer: `https://www.elastic.co/products/x-pack` redirects to `https://www.elastic.co/elastic-stack` now.
I don't think we need the italics or link here anymore as this is soon going away.

```suggestion
  which provides Ruby API for X-Pack APIs. This API is going to be merged into `elasticsearch-api` on v8.0.
```

> Please see their respective READMEs for information and documentation.

Reviewer: Not sure this is needed.
```suggestion
```

For integration with Ruby models and Rails applications,
see the **[elasticsearch-rails](https://github.com/elasticsearch/elasticsearch-rails)** project.

## Compatibility

> The Elasticsearch client is compatible with currently maintained Ruby versions. See [Ruby Maintenance Branches](https://www.ruby-lang.org/en/downloads/branches/) for more information. We don't provide support to versions which have reached their end of life.

Reviewer: Nit: `for more information` isn't needed.
```suggestion
The Elasticsearch client is compatible with currently maintained Ruby versions. See [Ruby Maintenance Branches](https://www.ruby-lang.org/en/downloads/branches/). We don't provide support to versions which have reached their end of life.
```

> The gem's version numbers follow the Elasticsearch major versions. The `master` branch is compatible with the Elasticsearch `master` branch, which is the next major version.

Reviewer: ```suggestion
The gem's version numbers follow Elasticsearch's major versions. The `master` branch is compatible with the Elasticsearch `master` branch, which is the next major version.
```
